### PR TITLE
Allow MultiApp cli_args to pass vectors to subapps

### LINKED ISSUE #20443: cli_args in MultiApp does not work for std::vector parameters
## Bug Description

```
cli_args = 'object/vec_param="1 2"'
```

are chopped at the space between `1` and `2` because `cli_args` is a `std::vector<std::string>`
parameter in MOOSE. That behavior is incorrect.

## Steps to Reproduce

See example above. 

## Impact

Prevents implementation of reasonable input handling for equilibrium pebble bed depletion.

### PR DESCRIPTION
**Pull request by snschune**
closes #20443
## Reason

MultiApp `cli_args` does not support vector parameters.

## Design

`cli_args` is now a `std::string` parameter that is parsed by a dedicated function. 

## Impact

No impact is anticipated.

### CODE REVIEW

**Comment by snschune** [comment]
I am breaking something in stochastic tools module that relies on `cli_args` to be `std::vector<string>`.
Maybe it's a better idea to have a template specialization for reading `std::vector<string>` that honors
spaces in quotes. What do you think @loganharbour, @zachmprince if there is a better way to address failures
in stochastic tools.

File: test/tests/multiapps/command_line/master.i [hunk 1]
```diff
@@ -18,6 +18,7 @@
     positions = '0 0 0
                  1 1 1'
     input_files = 'sub.i'
-    cli_args = 'Mesh/nx=10 Mesh/nx=100'
+    cli_args = 'Mesh/mesh/type=GeneratedMeshGenerator;Mesh/mesh/dim=1;Mesh/mesh/nx=10
```
Comment (YaqiWang) [inline test/tests/multiapps/command_line/master.i:21]: Should this be
```
Mesh/mesh/type=GeneratedMeshGenerator Mesh/mesh/dim=1 Mesh/mesh/nx=10;
Mesh/mesh/type=GeneratedMeshGenerator Mesh/mesh/dim=1 Mesh/mesh/nx=100
```

**Comment by GiudGiud** [comment]
I'll take this over after Positions are merged
I've changed this a bit for matching Positions & cli_args so the work will need to be adapted.

**Comment by GiudGiud** [comment]
I dont think this is an easy fix at this point. It s no longer a std::vector<std::string> and it s also no longer a reference or const. The latter is what s needed for controlling the parameter. 

Modifying the parsing seems like the most viable route here

File: test/tests/multiapps/command_line/master.i (hunk 1 above)
Comment (GiudGiud) [reply test/tests/multiapps/command_line/master.i:21]: I'm open to either but we need to decide. I m redoing the parsing of CLI argument strings

Comment (GiudGiud) [reply test/tests/multiapps/command_line/master.i:21]: old syntax with "space" to declare change of argument is more compatible with what we do elsewhere:
"space" for vector
; for vector of vector

**Comment by loganharbour** [comment]
yeah, this is going to take more work.

**Comment by GiudGiud** [comment]
@snschune

**Comment by snschune** [comment]
@GiudGiud I tested the changes and they work for my problem. Thanks.

**Comment by GiudGiud** [comment]
Thank you for checking

Reviewers please have at it

File: test/tests/multiapps/command_line/tests [hunk 2]
```diff
@@ -5,20 +5,63 @@
...
+                  "using double quotes, parameters for all sub applications from the input file."
+  []
+
+  [common_vector_single]
+    type = Exodiff
+    input = parent_common_vector.i
+    cli_args = 'MultiApps/active=sub_2'
+    exodiff = 'parent_common_vector_out_sub_20.e parent_common_vector_out_sub_21.e'
+    requirement = "The MultiApp system shall include the ability to set common command line vector, "
+                  "using single quotes, parameters for all sub applications from the input file."
```
Comment (lindsayad) [inline test/tests/multiapps/command_line/tests:34]: change wording, potentially as above

File: test/tests/multiapps/command_line/tests [hunk 3]
```diff
@@ -5,20 +5,63 @@
...
+    prereq = 'common_vector'
+  []
+
+  [split_with_vector_single]
+    type = Exodiff
+    input = parent_common_vector.i
+    cli_args = 'MultiApps/active=sub_1_split'
+    exodiff = 'parent_common_vector_out_sub_1_split0.e parent_common_vector_out_sub_1_split1.e'
+    requirement = "The MultiApp system shall include the ability to set command line vector, using "
+                  "double quotes, parameters distributed among all sub applications from the input "
```
Comment (lindsayad) [inline test/tests/multiapps/command_line/tests:44]: wording

**Review by lindsayad — CHANGES_REQUESTED** [review]

File: test/tests/multiapps/command_line/tests (hunk 2 above)
Comment (GiudGiud) [inline test/tests/multiapps/command_line/tests:34]:
```suggestion
    requirement = "The MultiApp system shall include the ability to set common command line vector parameters, "
                  "using single quotes, for all sub applications from the input file."
```

File: test/tests/multiapps/command_line/tests [hunk 4]
```diff
@@ -5,20 +5,63 @@
...
+    prereq = 'common_vector_single'
+  []
+
+  [split_with_vector_double]
+    type = Exodiff
+    input = parent_common_vector.i
+    cli_args = 'MultiApps/active=sub_2_split'
+    exodiff = 'parent_common_vector_out_sub_2_split0.e parent_common_vector_out_sub_2_split1.e'
+    requirement = "The MultiApp system shall include the ability to set command line vector, using "
+                  "single quotes, parameters distributed among all sub applications from the input "
```
Comment (GiudGiud) [inline test/tests/multiapps/command_line/tests:55]:
```suggestion
    requirement = "The MultiApp system shall include the ability to set command line vector parameters, "
                  "using single quotes, distributed among all sub applications from the input "
```

File: test/tests/multiapps/command_line/tests [hunk 5]
```diff
@@ -5,20 +5,63 @@
...
+    prereq = 'common_vector'
+  []
+
+  [split_with_vector_single]
+    type = Exodiff
+    input = parent_common_vector.i
+    cli_args = 'MultiApps/active=sub_1_split'
+    exodiff = 'parent_common_vector_out_sub_1_split0.e parent_common_vector_out_sub_1_split1.e'
+    requirement = "The MultiApp system shall include the ability to set command line vector parameters,"
+                  " using double quotes, distributed among all sub applications from the input "
```
Comment (lindsayad) [inline test/tests/multiapps/command_line/tests:44]:
```suggestion
    requirement = "The MultiApp system shall include the ability to set command line vector parameters, "
                  "using double quotes, distributed among all sub applications from the input "
```

**Comment by GiudGiud** [comment]
squashed

File: test/tests/multiapps/command_line/tests [hunk 6]
```diff
@@ -5,20 +5,63 @@
     type = Exodiff
     input = parent.i
     exodiff = 'parent_out_sub0.e parent_out_sub1.e'
-    requirement = "The MultiApp system shall include the ability to set command line parameters for each sub application from the input file."
+    requirement = "The MultiApp system shall include the ability to set command line parameters for "
```
Comment (lindsayad) [inline test/tests/multiapps/command_line/tests:8]: Boo at the use of an implementation detail but as we have published on this name for a decade and it is a part of what makes MOOSE MOOSE, then it seems fine, e.g. it's more of a colloquial noun at this point as opposed to an implementation noun

**Review by lindsayad — APPROVED** [review]

**Comment by GiudGiud** [comment]
Thanks for the review and thanks for the contribution @snschune !

**Pull request merged**